From b2c7dde9dbef324175d1c894681b39c56740b03d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 29 Aug 2022 21:26:49 +0300
Subject: [PATCH]  [SW-102303] gaudi3: stop timestamp before init

Before enabling timestamp as part of timestamp init, this
patch add changes to stop timestamp before initialization.

Change-Id: Ib60ace360ec81675098563f675feb317c6d48ee9
---

diff --git a/common/delay.c b/common/delay.c
index 01ff97c..7ccd99b 100644
--- a/common/delay.c
+++ b/common/delay.c
@@ -23,6 +23,9 @@
 
 	cycles_per_usec = timestamp_freq / USEC_PER_SEC;
 
+	/* stop counter */
+	regmap_writel(0x0, mmTIMESTAMP_CNTCR);
+
 	/* timer init */
 	regmap_writel(timestamp_freq, mmTIMESTAMP_CNTFID0);
 	regmap_writel((TIMESTAMP_CNTCR_CNTCR_EN_M | TIMESTAMP_CNTCR_CNTCR_HDBG_M),
